### PR TITLE
Purge inactive communities

### PR DESCRIPTION
**Pull request by NicJak**
* Purge inactive communities as required by #107

### CODE REVIEW

File: ceremonies/src/lib.rs [hunk 1]
```diff
@@ -575,6 +582,32 @@ impl<T: Config> Module<T> {
...
+					InactivityCounters::insert(cid, current + 1);
+				}
+			} else {
+				InactivityCounters::insert(cid, 0);
+			}
+		}
+	}
+
+	fn purge_community(cid: CommunityIdentifier) {
+		for cindex in 1..<encointer_scheduler::Module<T>>::current_ceremony_index() {
```
Comment (brenzi) [inline ceremonies/src/lib.rs:603]: No need to start with 1 here. Too expensive. the registry is purged after reputation lifetime anyway

File: ceremonies/src/lib.rs [hunk 2]
```diff
@@ -575,6 +582,32 @@ impl<T: Config> Module<T> {
...
+			if cindex > T::ReputationLifetime::get() {
```
Comment (brenzi) [inline ceremonies/src/lib.rs:604]: Db access within a loop. Too expensive

**Review by brenzi — CHANGES_REQUESTED** [review]
Please add unit tests asserting:
* an inactive community gets purged at the expected time
* an active community doesn' t get purged
* purging does remove all meetup locations correctly. And only the right local are purged

File: ceremonies/src/tests.rs [hunk 3]
```diff
@@ -1253,6 +1253,49 @@ fn get_assignment_params_works() {
...
+		let cid1 = CommunityIdentifier::new(
+			Location::new(Degree::from_num(1f64), Degree::from_num(1f64)),
+			Vec::<i64>::new(),
+		)
+		.unwrap();
+
+		let mut cindex = 5;
+		let assignment =
+			AssignmentCount { bootstrappers: 1, reputables: 0, endorsees: 0, newbies: 0 };
+		AssignmentCounts::insert((cid0, cindex), assignment);
```
Comment (brenzi) [inline ceremonies/src/tests.rs:1269]: #107 demands successful meetups. You are only checking assignment of meetups which might not actually have taken place. I guess you should check Reputation instead oft assignments. Or introduce a flag to signal at least one successful meetup has been verified and issued

**Review by brenzi — CHANGES_REQUESTED** [review]

File: ceremonies/src/tests.rs (hunk 3 above)
Comment (NicJak) [reply ceremonies/src/tests.rs:1269]: how about checking attestationcount!=0 of the last cindex during registration phase?

Comment (brenzi) [reply ceremonies/src/tests.rs]: No. Attestations must be validated. Can't be trusted

Comment (brenzi) [reply ceremonies/src/tests.rs]: Check Reputation registry or introduce a flag if that is too expensive

**Review by brenzi — APPROVED** [review]
LGTM

File: ceremonies/src/lib.rs [hunk 4]
```diff
@@ -766,37 +773,41 @@ impl<T: Config> Pallet<T> {
...
 
-			<IssuedRewards<T>>::remove_prefix((cid, cindex), None);
+		<IssuedRewards<T>>::remove_prefix(cc, None);
+	}
+
+	fn purge_registry(cindex: CeremonyIndexType) {
+		let cids = <encointer_communities::Pallet<T>>::community_identifiers();
+		for cid in cids.iter() {
+			Self::purge_community_ceremony((*cid, cindex));
 		}
```
Comment (clangenb) [inline ceremonies/src/lib.rs:811]: Even though, copy-dereferencing is already cheap, you can save a bit of efficiency by using `into_iter`, which returns owned values.

**Review by clangenb — APPROVED** [review]
Looks good to me!

Thanks for the minor cleanup!

There is one efficiency detail, but I think we can neglect that.

**Pull request merged**